src/Cake.Prca/Issues/StringPathExtensions.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Based on http://stackoverflow.com/a/31941159/566901
+
+namespace Cake.Prca.Issues
+{
+    using System;
+    using System.IO;
+    using System.Linq;
+
+    /// <summary>
+    /// Extensions for <see cref="string"/> for handling paths.
+    /// </summary>
+    public static class StringPathExtensions
+    {
+        /// <summary>
+        /// Checks if a string containing a path is a valid path string.
+        /// </summary>
+        /// <param name="path">Path to check.</param>
+        /// <returns><c>True</c> if valid path.</returns>
+        public static bool IsValidPath(this string path)
+        {
+            path.NotNullOrWhiteSpace(nameof(path));
+
+            return path.IndexOfAny(Path.GetInvalidPathChars().ToArray()) == -1;
+        }
+
+        /// <summary>
+        /// Checks if a string containing a path is a full path.
+        /// </summary>
+        /// <param name="path">Path which should be checked</param>
+        /// <returns><c>True</c> if full path.</returns>
+        public static bool IsFullPath(this string path)
+        {
+            path.NotNullOrWhiteSpace(nameof(path));
+
+            if (!path.IsValidPath())
+            {
+                throw new ArgumentException("Invalid path", nameof(path));
+            }
+
+            // ReSharper disable once PossibleNullReferenceException
+            return
+                Path.IsPathRooted(path) &&
+                !Path.GetPathRoot(path).Equals(Path.DirectorySeparatorChar.ToString(), StringComparison.Ordinal);
+        }
+
+        /// <summary>
+        /// Checks if a path is a sub path of another path.
+        /// The comparison is case-insensitive, handles <c>/</c> and <c>\</c> slashes as folder separators and
+        /// only matches if the base dir folder name is matched exactly
+        /// (<c>c:\foobar\file.txt</c> is not a sub path of <c>c:\foo</c>).
+        /// </summary>
+        /// <param name="path">Full path which should be checked if it is a sub path.</param>
+        /// <param name="baseDirPath">Path for which should be checked if <paramref name="path"/>is a sub path of.</param>
+        /// <returns>Returns true if <paramref name="path"/> starts with the path <paramref name="baseDirPath"/>.</returns>
+        public static bool IsSubpathOf(this string path, string baseDirPath)
+        {
+            path.NotNullOrWhiteSpace(nameof(path));
+            baseDirPath.NotNullOrWhiteSpace(nameof(baseDirPath));
+
+            if (!path.IsValidPath())
+            {
+                throw new ArgumentException("Invalid path", nameof(path));
+            }
+
+            if (!baseDirPath.IsValidPath())
+            {
+                throw new ArgumentException("Invalid path", nameof(baseDirPath));
+            }
+
+            var normalizedPath =
+                Path.GetFullPath(path.NormalizePath().WithEnding("\\"));
```

Comment:
Using GetFullPath here could lead to some wrong results, especially when the paths contains `..\`, which go beyond the number of segments of `Environment.CurrentDirectory`.

```cs
var path = "Test\\Test2";
var basePath = "Test";

if (!IsSubpathOf(path, basePath)) { throw new Exception(); }

if (IsSubpathOf("..\\" + path, basePath)) { throw new Exception(); }

for (int li = 0; li <  Environment.CurrentDirectory.Split('\\').Length + 1 ; li++)
{
	path = "..\\" + path;
	basePath = "..\\" + basePath;
}

if (!IsSubpathOf(path, basePath)) { throw new Exception(); }

if (IsSubpathOf("..\\" + path, basePath)) { throw new Exception("Should not happen."); }
```